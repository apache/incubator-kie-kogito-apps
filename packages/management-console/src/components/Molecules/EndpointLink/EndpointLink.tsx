import React from 'react';
import { Button } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';

interface IOwnProps {
  serviceUrl: string;
  isLinkShown: boolean;
}

const EndpointLink: React.FC<IOwnProps> = ({ serviceUrl, isLinkShown }) => {
  return (
    <>
      {serviceUrl !== null ? (
        <Button
          component={'a'}
          variant={'link'}
          target={'_blank'}
          href={`${serviceUrl}`}
          isInline={true}
        >
          {isLinkShown ? serviceUrl : 'Endpoint'}
          {<ExternalLinkAltIcon className="pf-u-ml-xs" />}
        </Button>
      ) : (
        ''
      )}
    </>
  );
};

export default EndpointLink;
