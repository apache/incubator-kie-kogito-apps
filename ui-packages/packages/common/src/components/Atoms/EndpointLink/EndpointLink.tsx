import React from 'react';
import { Button } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import { OUIAProps, componentOuiaProps } from '../../../utils/OuiaUtils';

interface IOwnProps {
  serviceUrl: string;
  isLinkShown: boolean;
}

const EndpointLink: React.FC<IOwnProps & OUIAProps> = ({
  serviceUrl,
  isLinkShown,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <>
      {serviceUrl !== null ? (
        <Button
          component={'a'}
          variant={'link'}
          target={'_blank'}
          href={`${serviceUrl}`}
          isInline={true}
          {...componentOuiaProps(ouiaId, 'endpoint-link', ouiaSafe)}
        >
          {isLinkShown ? serviceUrl : 'Endpoint'}
          {<ExternalLinkAltIcon className="pf-u-ml-xs" />}
        </Button>
      ) : (
        ''
      )}
    </>
  );
};

export default EndpointLink;
