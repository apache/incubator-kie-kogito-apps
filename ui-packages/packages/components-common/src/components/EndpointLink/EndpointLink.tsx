import React from 'react';
import { Button } from '@patternfly/react-core/dist/js/components/Button';
import { ExternalLinkAltIcon } from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
import {
  componentOuiaProps,
  OUIAProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';
interface IOwnProps {
  serviceUrl: string;
  isLinkShown: boolean;
  linkLabel?: string;
}

export const EndpointLink: React.FC<IOwnProps & OUIAProps> = ({
  serviceUrl,
  isLinkShown,
  linkLabel,
  ouiaId,
  ouiaSafe
}) => {
  if (serviceUrl) {
    return (
      <Button
        component={'a'}
        variant={'link'}
        target={'_blank'}
        href={`${serviceUrl}`}
        isInline={true}
        {...componentOuiaProps(ouiaId, 'endpoint-link', ouiaSafe)}
      >
        {isLinkShown ? serviceUrl : linkLabel || 'Endpoint'}
        {<ExternalLinkAltIcon className="pf-u-ml-xs" />}
      </Button>
    );
  } else {
    return <></>;
  }
};
