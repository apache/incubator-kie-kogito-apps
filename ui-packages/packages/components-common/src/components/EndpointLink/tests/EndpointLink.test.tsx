/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { EndpointLink } from '../EndpointLink';

const props1 = {
  serviceUrl: 'http://localhost:4000/',
  isLinkShown: true,
  ouiaId: 'endpoint-link-1'
};

const props2 = {
  serviceUrl: 'http://localhost:4000/',
  isLinkShown: false,
  ouiaId: 'endpoint-link-2'
};

const props3 = {
  serviceUrl: 'http://localhost:4000/',
  isLinkShown: false,
  linkLabel: 'This is a label',
  ouiaId: 'endpoint-link-3'
};

const props4 = {
  serviceUrl: null,
  isLinkShown: false,
  ouiaId: 'endpoint-link-4'
};
describe('EndpointLink component tests', () => {
  it('snapshot testing for link shown', () => {
    const { container } = render(<EndpointLink {...props1} />);
    expect(container).toMatchSnapshot();
  });
  it('snapshot testing for link hidden', () => {
    const { container } = render(<EndpointLink {...props2} />);
    expect(container).toMatchSnapshot();
  });
  it('snapshot testing for link hidden with custom link label', () => {
    const { container } = render(<EndpointLink {...props3} />);
    expect(container).toMatchSnapshot();
  });
  it('snapshot testing no service URL and link hidden', () => {
    const { container } = render(<EndpointLink {...props4} />);
    expect(container).toMatchSnapshot();
  });
});
