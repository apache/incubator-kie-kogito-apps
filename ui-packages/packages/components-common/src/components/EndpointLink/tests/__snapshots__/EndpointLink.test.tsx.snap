// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndpointLink component tests snapshot testing for link hidden 1`] = `
<Button
  component="a"
  data-ouia-component-id="endpoint-link-2"
  data-ouia-component-type="endpoint-link"
  data-ouia-safe={true}
  href="http://localhost:4000/"
  isInline={true}
  target="_blank"
  variant="link"
>
  Endpoint
  <ExternalLinkAltIcon
    className="pf-u-ml-xs"
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
  />
</Button>
`;

exports[`EndpointLink component tests snapshot testing for link hidden with custom link label 1`] = `
<Button
  component="a"
  data-ouia-component-id="endpoint-link-3"
  data-ouia-component-type="endpoint-link"
  data-ouia-safe={true}
  href="http://localhost:4000/"
  isInline={true}
  target="_blank"
  variant="link"
>
  This is a label
  <ExternalLinkAltIcon
    className="pf-u-ml-xs"
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
  />
</Button>
`;

exports[`EndpointLink component tests snapshot testing for link shown 1`] = `
<Button
  component="a"
  data-ouia-component-id="endpoint-link-1"
  data-ouia-component-type="endpoint-link"
  data-ouia-safe={true}
  href="http://localhost:4000/"
  isInline={true}
  target="_blank"
  variant="link"
>
  http://localhost:4000/
  <ExternalLinkAltIcon
    className="pf-u-ml-xs"
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
  />
</Button>
`;

exports[`EndpointLink component tests snapshot testing no service URL and link hidden 1`] = `<Fragment />`;
